pkg/trace/api/telemetry_test.go

```diff
@@ -0,0 +1,204 @@
+	return httptest.NewTLSServer(http.HandlerFunc(func(w http.ResponseWriter, req *http.Request) {
+		body, err := ioutil.ReadAll(req.Body)
+		assert.NoError(t, err)
+		assert.NoError(t, onReq(req, body))
+		_, err = w.Write([]byte("OK"))
+		assert.NoError(t, err)
+		req.Body.Close()
+	}))
+}
+
+func newRequestRecorder(t *testing.T) (req *http.Request, rec *httptest.ResponseRecorder) {
+	req, err := http.NewRequest("POST", "/telemetry/proxy/path", strings.NewReader("body"))
+	assert.NoError(t, err)
+	rec = httptest.NewRecorder()
+	return req, rec
+}
+
+func recordedResponse(t *testing.T, rec *httptest.ResponseRecorder) string {
+	responseBody, err := ioutil.ReadAll(rec.Result().Body)
+	assert.NoError(t, err)
+
+	return string(responseBody)
+}
+
+func loadConfig(t *testing.T) *traceconfig.AgentConfig {
+	cfg, err := traceconfig.Load("/does/not/exists.yaml")
+	assert.NoError(t, err)
+	return cfg
+}
+
+func TestTelemetryBasicProxyRequest(t *testing.T) {
+	var endpointCalled uint64
+	assert := assert.New(t)
+
+	srv := assertingServer(t, func(req *http.Request, body []byte) error {
+		assert.Equal("body", string(body), "invalid request body")
+		assert.Equal("test_apikey", req.Header.Get("DD-API-KEY"))
+		assert.Equal("test_hostname", req.Header.Get("DD-Agent-Hostname"))
+		assert.Equal("test_env", req.Header.Get("DD-Agent-Env"))
+		assert.Equal("/path", req.URL.Path)
+
+		atomic.AddUint64(&endpointCalled, 1)
+		return nil
+	})
+	defer mockConfigMap(map[string]interface{}{
+		"api_key":                     "test_apikey",
+		"apm_config.telemetry.dd_url": srv.URL,
+		"hostname":                    "test_hostname",
+		"skip_ssl_validation":         true,
+		"env":                         "test_env",
+	})() // reset config after the test
+
+	req, rec := newRequestRecorder(t)
+	recv := NewHTTPReceiver(loadConfig(t), nil, nil, nil)
+	recv.buildMux().ServeHTTP(rec, req)
+
+	assert.Equal("OK", recordedResponse(t, rec))
+	assert.Equal(uint64(1), atomic.LoadUint64(&endpointCalled))
+}
+
+func TestTelemetryProxyMultipleEndpoints(t *testing.T) {
+	var endpointCalled uint64
+	assert := assert.New(t)
+
+	mainBackend := assertingServer(t, func(req *http.Request, body []byte) error {
+		assert.Equal("body", string(body), "invalid request body")
+		assert.Equal("test_apikey_1", req.Header.Get("DD-API-KEY"))
+		assert.Equal("test_hostname", req.Header.Get("DD-Agent-Hostname"))
+		assert.Equal("test_env", req.Header.Get("DD-Agent-Env"))
+
+		atomic.AddUint64(&endpointCalled, 2)
+		return nil
+	})
+	additionalBackend := assertingServer(t, func(req *http.Request, body []byte) error {
+		assert.Equal("body", string(body), "invalid request body")
+		assert.Equal("test_apikey_2", req.Header.Get("DD-API-KEY"))
+		assert.Equal("test_hostname", req.Header.Get("DD-Agent-Hostname"))
+		assert.Equal("test_env", req.Header.Get("DD-Agent-Env"))
+
+		atomic.AddUint64(&endpointCalled, 3)
+		return nil
+	})
+
+	additionalEndpoints := make(map[string]string)
```

Comment:
Do you mind removing this variable and defining the map inline below inside `mockConfigMap`? It's easier to read and follow. Plus, the variable is not really used anywhere else.

Reply:
Yeah, it turned out I assumed wrong about what are the properties of Go maps. Your suggested way is much better 🙇